### PR TITLE
Support CDATA for textual data

### PR DESCRIPTION
**Pull request by JemDay**
Signed-off-by: Jem Day <Jem.Day@cliffhanger.com>

Fixes: #1013 

## Proposed Changes

- Update XML Format specification to include support for CDATA textual content.

**Release Note**

```release-note
Allows for textual content in an XML Format to be carried in a CDATA section if preferred.

```

### CODE REVIEW

File: cloudevents/working-drafts/xml-format.md [hunk 1]
```diff
@@ -135,19 +135,35 @@ Example:
 
 ### 3.2 Text Data
 
-Text MUST be carried in an element with a defined type of `xs:string`.
-The element MUST NOT contain any child elements.
+Text MUST be carried in an element with a defined type of `xs:string`. The content
```
Comment (duglin) [inline cloudevents/working-drafts/xml-format.md:138]: Minor thing, the doc seems to switch between "xs" and "xsi" - we mean the same thing in both case right? Should we pick one for consistency?

**Comment by duglin** [comment]
LGTM

File: cloudevents/working-drafts/xml-format.md [hunk 2]
```diff
@@ -135,19 +135,35 @@ Example:
...
+of this element is either:
+
+* Regular XML textual content, or
+* A CDATA section.
+
+Implementations MUST accept both textual representations.
```
Comment (jskeet) [inline cloudevents/working-drafts/xml-format.md:144]: This currently seems to restrict it to one or the other - I *suspect* a mixture makes sense, e.g.

```xml
<data xsi:type="xs:string">Text before <![CDATA[if X < Y { .. }]]> Text after</data>
```

Something like "The content of this element is text, which may be expressed as any combination of text nodes and CDATA nodes. Implementations MUST accept all combinations."

(Not sure about the "all combinations" it, but we can noodle on that further.)

**Review by jskeet — COMMENTED** [review]
Definitely happy with the goal of this - just a nit on precisely what's allowed. It could be that I'm missing something about CDATA that makes combinations invalid in XML anyway. (It's definitely not an area I know much about.)

**Comment by duglin** [comment]
@JemDay and @jskeet will think about this one off-line - per the 1/26 call

**Comment by duglin** [comment]
@JemDay @jskeet what's the status of this one? Did you guys get a chance to chat?

**Comment by jskeet** [comment]
Still pending, basically. I think the action is probably on me to come up with a proposal, then send it to Jem.

**Comment by jskeet** [comment]
(On that front, if I could be given sufficient permissions to assign PR reviewers etc, that would be handy. Not to worry if you'd prefer not to though.)

**Comment by duglin** [comment]
@jskeet give it a try now

**Comment by jskeet** [comment]
> @jskeet give it a try now

Excellent, thanks :)

**Comment by duglin** [comment]
just a reminder... rebase and action needed

**Comment by jskeet** [comment]
Won't bother rebasing until we've worked out what we're going to do. No progress yet, I'm afraid.

**Comment by JemDay** [comment]
Thanks @duglin - I think we're good.

**Pull request closed without merging**